### PR TITLE
CB-16446 add in recovery option for detach

### PR DESCRIPTION
**Pull request by yg-gd**
Expands on work done in CB-14399 needs to plan action based on current detached status

### CODE REVIEW

File: datalake/src/main/java/com/sequenceiq/datalake/service/resize/recovery/ResizeRecoveryService.java [hunk 1]
```diff
@@ -90,6 +96,13 @@ public SdxRecoveryResponse triggerRecovery(SdxCluster sdxCluster, SdxRecoveryReq
             switch (actualStatusForSdx.getStatus()) {
                 case STOP_FAILED:
                     return new SdxRecoveryResponse(sdxReactorFlowManager.triggerSdxStartFlow(sdxCluster));
+                case STOPPED:
+                    if (sdxCluster.isDetached()) {
+                        //return new SdxRecoveryResponse(sdxReactorFlowManager.triggerSdxResizeRecovery(sdxCluster, null));
+                        throw new NotImplementedException("Cluster is currently in an unrecoverable state");
```
Comment (sxxgrc) [inline datalake/src/main/java/com/sequenceiq/datalake/service/resize/recovery/ResizeRecoveryService.java:102]: Reattaching can technically be done via the recovery flow chain but there has definitely not been enough testing to ensure it covers all possible cases.

I think this is good for now, but in the future we may need to extend this to cover that too. Would be nice to have as much recovery as possible via the UI/CLI.

**Review by sxxgrc — APPROVED** [review]

File: datalake/src/main/java/com/sequenceiq/datalake/service/resize/recovery/ResizeRecoveryService.java (hunk 1 above)
Comment (bergerdenes) [inline datalake/src/main/java/com/sequenceiq/datalake/service/resize/recovery/ResizeRecoveryService.java:102]: this contradicts with line#75 where the validity check returns RECOVERABLE

**Review by bergerdenes — CHANGES_REQUESTED** [review]

File: datalake/src/main/java/com/sequenceiq/datalake/service/resize/recovery/ResizeRecoveryService.java [hunk 2]
```diff
@@ -70,6 +70,12 @@ public SdxRecoverableResponse validateRecovery(SdxCluster sdxCluster) {
         switch (actualStatusForSdx.getStatus()) {
             case STOP_FAILED:
                 return new SdxRecoverableResponse("Resize can be recovered from a failed stop", RecoveryStatus.RECOVERABLE);
+            case STOPPED:
+                if (sdxCluster.isDetached()) {
+                    return new SdxRecoverableResponse("Resize can reattach cluster", RecoveryStatus.RECOVERABLE);
```
Comment (kkalvagadda1) [inline datalake/src/main/java/com/sequenceiq/datalake/service/resize/recovery/ResizeRecoveryService.java:75]: Until we test it and are sure of it, let's call this non-recoverable state.

When the changes for [CB-14399](https://jira.cloudera.com/browse/CB-14399) land. We can very the behavior and make changes accordingly.

**Review by kkalvagadda1 — CHANGES_REQUESTED** [review]

**Review by kkalvagadda1 — APPROVED** [review]

File: datalake/src/main/java/com/sequenceiq/datalake/service/resize/recovery/ResizeRecoveryService.java (hunk 1 above)
Comment (doktoric) [reply datalake/src/main/java/com/sequenceiq/datalake/service/resize/recovery/ResizeRecoveryService.java:102]: @yg-gd is this the same as line #75 right?

Comment (yg-gd) [reply datalake/src/main/java/com/sequenceiq/datalake/service/resize/recovery/ResizeRecoveryService.java:102]: correct now Non-Recoverable we shouldn't run into this

**Review by bergerdenes — APPROVED** [review]

**Pull request merged**